piecrust/baking/single.py

```diff
@@ -82,7 +84,8 @@ def bake(self, qualified_page, prev_entry, dirty_source_names,
             # Get the URL and path for this sub-page.
             sub_uri = qualified_page.getUri(cur_sub)
             logger.debug("Baking '%s' [%d]..." % (sub_uri, cur_sub))
-            out_path = self.getOutputPath(sub_uri)
+            pretty_urls = qualified_page.pretty_urls if qualified_page.pretty_urls is not None else self.pretty_urls
```

Comment:
Based on my other comment below, you can replace this with just: `pretty_urls = qualified_page.get('pretty_urls')`. The rest should work fine since you're already falling back to `self.pretty_urls` inside of `getOutputPath` if the optional argument isn't there.